Sources/TuistDependencies/Carthage/Interactors/CarthageFrameworksInteractor.swift

```diff
@@ -0,0 +1,103 @@
+
+            try builtFrameworks.forEach { frameworkName in
+                let carthageBuildFrameworkPath = carthagePlatfromBuildsDirectory.appending(component: "\(frameworkName).framework")
+                let destinationFramemorekPath = dependenciesDirectory.appending(components: frameworkName, platform.caseValue, "\(frameworkName).framework")
+                try copyDirectory(from: carthageBuildFrameworkPath, to: destinationFramemorekPath)
+            }
+
+            let existingFrameworks: Set<String> = Set(graph.dependencies(for: platform))
+            let frameworksToDelete = existingFrameworks.subtracting(builtFrameworks)
+
+            try frameworksToDelete.forEach { frameworkName in
+                let destinationFrameworkPath = dependenciesDirectory.appending(components: frameworkName, platform.caseValue, "\(frameworkName).framework")
+                try deleteDirectory(at: destinationFrameworkPath)
+            }
+
+            newGraph = newGraph.updatingDependencies(Array(builtFrameworks), for: platform)
+        }
+
+        try saveGraph(graph: newGraph, graphPath: graphPath)
+    }
+
+    // MARK: - Helpers
+
+    private func buildCarthagePlatfromBuildsDirectory(carthageBuildDirectory: AbsolutePath, platform: Platform) -> AbsolutePath {
+        switch platform {
+        case .iOS, .watchOS, .tvOS:
+            return carthageBuildDirectory.appending(component: platform.caseValue)
+        case .macOS:
+            return carthageBuildDirectory.appending(component: "Mac")
+        }
+    }
+
+    private func getBuiltFrameworks(carthagePlatfromBuildsDirectory: AbsolutePath) throws -> [String] {
+        try fileHandler
+            .contentsOfDirectory(carthagePlatfromBuildsDirectory)
+            .filter { $0.isFolder && $0.extension == "framework" }
+            .compactMap { $0.components.last?.components(separatedBy: ".").first }
+    }
+
+    private func readGraph(graphPath: AbsolutePath) -> Graph {
+        do {
+            let decoder = JSONDecoder()
+            let graphFileData = try fileHandler.readFile(graphPath)
+            return try decoder.decode(Graph.self, from: graphFileData)
+        } catch {
+            return .empty
+        }
+    }
+
+    private func saveGraph(graph: Graph, graphPath: AbsolutePath) throws {
+        let encoder = JSONEncoder()
+        let graphFileData = try encoder.encode(graph)
+        try fileHandler.write(String(data: graphFileData, encoding: .utf8) ?? "", path: graphPath, atomically: true)
+    }
+
+    private func copyDirectory(from fromPath: AbsolutePath, to toPath: AbsolutePath) throws {
+        try fileHandler.createFolder(toPath.removingLastComponent())
+
+        if fileHandler.exists(toPath) {
+            try fileHandler.delete(toPath)
+        }
+
+        try fileHandler.copy(from: fromPath, to: toPath)
+    }
+
+    private func deleteDirectory(at path: AbsolutePath) throws {
```

Comment:
This method is only used from once place and it's 3 lines of code so I think we can remove it and move the code to the place where the method is called from.